### PR TITLE
feat(INVT-CPC-1153): Searching products status

### PR DESCRIPTION
**Pull request by CharlesS42**
**JIRA:** link to jira ticket
## Context:
Before, the front-end was capable of filtering the status of the products of an inventory, but not the back-end. Now, the back-end method meant to search products inside of an inventory is supporting the filtering by name, description and status of products. This change was necessary for later development where the search product feature will be exploited outside of the current inventory products page.
## Does this PR change the .vscode folder in petclinic-frontend?:
No.

<span style="color:red">**Reviewers need to check for any changes to the 
.vscode folder and add a comment about it to their review comments.**</span>

## Changes
Back-end method meant to search products inside of an inventory is supporting the filtering by name, description and status of products.
Increase testing coverage in the inventory-service and the api-gateway for the searchProducts method.

### CODE REVIEW

File: api-gateway/src/test/java/com/petclinic/bffapigateway/domainclientlayer/InventoryServiceClientIntegrationTest.java
Comment (FelixAllard) [inline api-gateway/src/test/java/com/petclinic/bffapigateway/domainclientlayer/InventoryServiceClientIntegrationTest.java:1]: Good lord! Good job with the testing!

File: inventory-service/src/main/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceImpl.java
Comment (FelixAllard) [inline inventory-service/src/main/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceImpl.java:1]: I see you added your status everywhere, well played!

File: inventory-service/src/test/java/com/petclinic/inventoryservice/datalayer/Product/ProductRepositoryTest.java
Comment (FelixAllard) [inline inventory-service/src/test/java/com/petclinic/inventoryservice/datalayer/Product/ProductRepositoryTest.java:1]: Once again, the status is there!

File: petclinic-frontend/src/features/products/components/ProductSearch.tsx
Comment (FelixAllard) [inline petclinic-frontend/src/features/products/components/ProductSearch.tsx:1]: How did the .ts got forgotten before 😭 . Anyway good notice

File: api-gateway/src/test/java/com/petclinic/bffapigateway/domainclientlayer/InventoryServiceClientIntegrationTest.java
Comment (Hares-2088) [inline api-gateway/src/test/java/com/petclinic/bffapigateway/domainclientlayer/InventoryServiceClientIntegrationTest.java:1]: Well done. those tests cover every situation

File: inventory-service/src/main/java/com/petclinic/inventoryservice/datalayer/Product/ProductRepository.java
Comment (Hares-2088) [inline inventory-service/src/main/java/com/petclinic/inventoryservice/datalayer/Product/ProductRepository.java:1]: I think you just did like 4 stories in one. Good job

**Review by Hares-2088 — APPROVED** [review]
Well done

**Review by FelixAllard — APPROVED** [review]
Everything looks fine to me! The status was added everywhere!

**Review by Carlos-123321 — COMMENTED** [review]
The highlight of this pr is probably the testing 😲. Good job and congrats on finishing pet clinic!🥳

**Pull request merged**